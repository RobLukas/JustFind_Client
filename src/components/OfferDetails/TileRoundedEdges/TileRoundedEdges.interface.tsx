import { ReactChild } from 'react';

interface TileRoundedEdgesProps {
  children: ReactChild;
}

export default TileRoundedEdgesProps;
