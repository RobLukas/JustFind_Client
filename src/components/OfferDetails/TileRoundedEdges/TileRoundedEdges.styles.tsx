import styled from 'styled-components';

const TileRoundedEdges = styled.span`
  @media (max-width: 1020px) {
    margin-left: auto;
  }
  color: ${({ theme }) => theme.styles.primary.color};
  border: 1px solid ${({ theme }) => theme.styles.offers.techTitle.border};
  background-color: ${({ theme }) => theme.styles.offers.techTitle.background};
  display: inline-block;
  margin-left: 6px;
  border-radius: 16px;
  text-transform: lowercase;
  white-space: nowrap;
  padding: 4px 7px;
  text-overflow: ellipsis;
  overflow: hidden;
  min-width: 1ch;
`;

export default { TileRoundedEdges };
