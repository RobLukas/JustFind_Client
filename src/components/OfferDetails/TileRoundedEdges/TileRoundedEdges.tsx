import React, { FC } from 'react';
import TileRoundedEdgesProps from './TileRoundedEdges.interface';
import S from './TileRoundedEdges.styles';

const TileRoundedEdges: FC<TileRoundedEdgesProps> = ({
  children,
}: TileRoundedEdgesProps) => {
  return <S.TileRoundedEdges>{children}</S.TileRoundedEdges>;
};

export default TileRoundedEdges;
