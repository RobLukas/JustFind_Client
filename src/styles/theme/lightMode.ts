export default {
  name: 'Light Theme',
  styles: {
    primary: {
      background: '#f3f6f8',
      color: '#37474f',
    },
    secondary: {
      background: '#fff',
      color: '#99a1ab',
    },
    dropdown: {
      button: {
        hover: '#f3f6f8',
      },
    },
    scrollbar: {
      thumbBackground: '#c1c1c1',
      background: 'rgba(0,0,0,0.07)',
      hover: 'rgba(0,0,0,0.09)',
    },
    filter: {
      button: {
        color: '#777',
      },
      selected: {
        background: 'linear-gradient(to right, #ba68c8, #9c27b0)',
        color: '#fff',
      },
    },
    offer: {
      header: {
        title: '#fff',
        filter: {
          brightness: '1',
        },
        icons: {
          companyName: '#ff5252',
          companySize: '#fb8c00',
          companyEMPType: '#ab47bc',
          companySkill: '#66bb6a',
          new: '#448aff',
        },
      },
      desc: {
        seperateLine: {
          color: '#f3f6f8',
        },
        skillPoint: {
          selected: '#ff4081',
          unselected: '#393939',
        },
      },
    },
    offers: {
      newTitle: {
        background: '#d9ddfc',
        color: '#4256ef',
      },
      techTitle: {
        background: '#fff',
        border: '#e0e0e0',
      },
    },
  },
};
