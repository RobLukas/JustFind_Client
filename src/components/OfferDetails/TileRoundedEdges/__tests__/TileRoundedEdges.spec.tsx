import React from 'react';
import { cleanup, render } from '@testing-library/react';
import 'jest-styled-components';

import lightMode from 'styles/theme/lightMode';
import { ThemeProvider } from 'styled-components';
import TileRoundedEdges from '../TileRoundedEdges';
import TileRoundedEdgesProps from '../TileRoundedEdges.interface';

describe('<TileRoundedEdges />', () => {
  afterEach(cleanup);
  it('render correctly', () => {
    const { children }: TileRoundedEdgesProps = { children: 'html' };

    const { container } = render(
      <ThemeProvider theme={lightMode}>
        <TileRoundedEdges>{children}</TileRoundedEdges>
      </ThemeProvider>,
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
