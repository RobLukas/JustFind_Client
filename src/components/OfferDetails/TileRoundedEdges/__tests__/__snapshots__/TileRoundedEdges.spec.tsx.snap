// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TileRoundedEdges /> render correctly 1`] = `
<span
  class="sc-bdnylx gXaVhD"
>
  html
</span>
`;
